pennylane_lightning/core/src/simulators/lightning_kokkos/gates/MatrixGateFunctors.hpp

```diff
@@ -104,6 +111,71 @@ template <class Precision> struct multiQubitOpFunctor {
     }
 };
 
+template <class Precision> struct NCMultiQubitOpFunctor {
```

Comment:
This construction is similar to [`multiQubitOpFunctor`](https://github.com/PennyLaneAI/pennylane-lightning/blob/b491a37635145449a88e68910794179f72a6ee6d/pennylane_lightning/core/src/simulators/lightning_kokkos/gates/MatrixGateFunctors.hpp#L35)